### PR TITLE
fix/docs: Update docs to reflect returned object from genAuth functions

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: src/native/_mutable.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ const MDataInfo = Struct({
   /// Name of the mutable data.
   name: t.XOR_NAME,
   /// Type tag of the mutable data.
-  type_tag: t.u64,
+  typeTag: t.u64,
```
Comment (bochaco) [inline src/native/_mutable.js]: I'm not sure if it really matters but perhaps is worthy keeping the prop name here as `type_tag` as it's supposed to match the native struct.

File: src/native/_mutable.js [hunk 2]
```diff
@@ -105,7 +105,7 @@ const makeMDataInfo = (mDataInfoObj) => {
   }
   return new MDataInfo({
     name: mDataInfoObj.name,
-    type_tag: mDataInfoObj.type_tag,
+    typeTag: mDataInfoObj.typeTag,
```
Comment (bochaco) [inline src/native/_mutable.js]: I'm not sure if it really matters but perhaps is worthy keeping the prop name here as `type_tag` as it's supposed to match the native struct, and only call it `typeTag` on the json we expose only

File: src/native/_mutable.js [hunk 3]
```diff
@@ -224,8 +224,8 @@ const makeMDataInfoObj = (mDataInfo) => {
   } catch (err) {
     throw makeError(errConst.XOR_NAME.code, errConst.XOR_NAME.msg(t.XOR_NAME.size));
   }
-  const type_tag = mDataInfo.type_tag;
-  if(!Number.isInteger(type_tag)) throw makeError(errConst.TYPE_TAG_NAN.code, errConst.TYPE_TAG_NAN.msg); 
+  const typeTag = mDataInfo.typeTag;
```
Comment (bochaco) [inline src/native/_mutable.js]: This should probably be `const typeTag = mDataInfo.type_tag;`

File: src/native/_auth.js [hunk 4]
```diff
@@ -43,7 +43,7 @@ const ContainerPermissionsArray = new ArrayType(ContainerPermissions);
 
 /// For use in `ShareMDataReq`. Represents a specific `MutableData` that is being shared.
 const ShareMData = Struct({
-  type_tag: t.u64,
+  typeTag: t.u64,
```
Comment (bochaco) [inline src/native/_auth.js]: I'm not sure if it really matters but perhaps is worthy keeping the prop name here as type_tag as it's supposed to match the native struct.

File: src/native/_auth.js [hunk 5]
```diff
@@ -277,7 +277,7 @@ const makeShareMDataPermissions = (permissions) => {
   return new ShareMDataArray(permissions.map((perm) => {
     const permsArray = helpers.makePermissionSet(perm.perms);
     return ShareMData({
-      type_tag: perm.type_tag,
+      typeTag: perm.typeTag,
```
Comment (bochaco) [inline src/native/_auth.js]: I think here we also need to keep `typeTag: perm.typeTag,` as I think both structs (in and out) are used by ffi, i.e. they are internal not a json we expose.

File: src/native/_mutable.js (hunk 3 above)
Comment (joshuef) [reply src/native/_mutable.js]: I think this one stands, as the `mDataInfo` is a js construct from line 108 above,.

**Comment by joshuef** [comment]
it's looking like the naming just needs to be consistent, and seems like the native FFI doesn't really mind what it's called (having some of one and some of another is causing test failures, wheres straight up swap seems fine)

**Pull request merged**